hack/kind-up.sh

```diff
@@ -151,6 +151,31 @@ EOF
 
 parse_flags "$@"
 
+# Prepare garden.local.gardener.cloud hostname that can be used everywhere to talk to the garden cluster.
+# Historically, we used the docker container name for this, but this differs between clusters with different names
+# and doesn't work in IPv6 kind clusters: https://github.com/kubernetes-sigs/kind/issues/3114
+# Hence, we "manually" inject a host configuration into the cluster that always resolves to the kind container's IP,
+# that serves our garden cluster API.
+# This works in
+# - the first and the second kind cluster
+# - in IPv4 and IPv6 kind clusters
+# - in ManagedSeeds
+
+garden_cluster="$CLUSTER_NAME"
+if [[ "$CLUSTER_NAME" == "gardener-local2" ]] ; then
+  # garden-local2 is used as a second seed cluster, the first kind cluster runs the gardener control plane
+  garden_cluster="gardener-local"
+fi
+
+if [[ "$CLUSTER_NAME" == "gardener-local2-ha-single-zone" ]]; then
+  garden_cluster="gardener-local-ha-single-zone"
+fi
+
+if [[ "$CLUSTER_NAME" == "gardener-extensions" ]] && [[ -n "$(docker ps -aq -f name="$garden_cluster"-control-plane)" ]]; then
+    docker start "$garden_cluster"-control-plane
+    exit 0
```

Comment:
Isn't it better to use `"${garden_cluster}-control-plane"`? Because right now it is possible for [Filename expansion](https://www.gnu.org/software/bash/manual/html_node/Filename-Expansion.html) to occur
Example:
```
$ touch example-o
$ garden_cluster=example
$ docker start "$garden_cluster"-[control-plane]
Error response from daemon: No such container: example-o
Error: failed to start containers: example-o
$ docker start "${garden_cluster}-[control-plane]"
Error response from daemon: No such container: example-[control-plane]
Error: failed to start containers: example-[control-plane]
```
In our concrete case it won't happen because there are no *, ?, [, ] symbols but if we use a mixture of `"${foo}bar"` and `"$foo"bar` in our scripts other developers are more likely to use `"$foo"bar` because it's shorter and simpler to write. And this will increase the probability of hitting a bug where `"$foo"bar`  will cause unexpected filename expansion